lib/adb-device.js

```diff
@@ -190,6 +190,64 @@ class ADBDevice {
     return data;
   }
 
+  // Receive a file listing and parse it into an array of file objects.
+  async recvList(localId, remoteId) {
+    // This is the structure of each directory entry returned by LIST
+    // see https://android.googlesource.com/platform/system/core/+/master/adb/SYNC.TXT#39
+    const props = [
+      'cmd',
+      'mode',
+      'size',
+      'modified',
+      'length',
+      'filename'
+    ];
+    const files = [];
+    let file = {};
+    let i = 0;
+
+    // Keep requesting new packets until we receive 'DONE'
+    // TODO: Do we need a timeout in case 'DONE' is never sent? Shouldn't happen
```

Comment:
There's a timeout set on the transfer calls in USB device, I think it's currently 1s or 2s. I've never had an issue receiving `DONE`, if there's an issue usually you'll get a `FAIL` before the start of the data transfer for your command. I think this is safe for now.